src/translator/byte_array_util.h

```diff
@@ -5,7 +5,7 @@ namespace marian {
 namespace bergamot {
 
 AlignedMemory loadFileToMemory(const std::string& path, size_t alignment);
-std::vector<std::shared_ptr<AlignedMemory>> getModelMemoryFromConfig(marian::Ptr<marian::Options> options);
+std::vector<AlignedMemory> getModelMemoryFromConfig(marian::Ptr<marian::Options> options);
```

Comment:
Why were those shared to begin with?

Is this the design envisioning models would persist in memory after being unloaded in case an async service call has not completed its promise?

Reply:
I discussed it here https://github.com/browsermt/bergamot-translator/pull/450#discussion_r1278586467 a bit.

Basically, I saw the structure for vocabulary and assumed someone knew something about the lifetime of these objects that I didn't. But looking closer, `shared_ptr` for vocabularies exists to avoid a second allocation of memory when you have joint vocabs (which we do more often than not). 

I'm not fully clear on that. The TranslationModel owns the memory for the models/vocab/shortlist, but exposes does expose raw pointers to the marian backend. But given the AsyncService creates the TranslationModel instance, I'm not sure in what instances it would be unloaded before the expected fulfilment of a promise.